app/scripts/minmax/dimMinMax.controller.js

```diff
@@ -169,27 +177,190 @@
+                var total = 0;
+                _.each(set.armor, function(armor) {
+                    var stat = _.findWhere(armor.item.normalStats, {statHash: hash}) || {scaled: 0, bonus: 0};
+                    total += stat.scaled + (armor.bonus_type == target_type ? stat.bonus : 0);
+                });
+                return total;
+            };
+            
+            function step(activeGaurdian, h, g, c, l, ci, gh, ar, processed_count) {
+                for(; h < helms.length; ++h) {
+                for(; g < gaunts.length; ++g) {
+                for(; c < chests.length; ++c) {
+                for(; l < legs.length; ++l) {
+                for(; ci < classItems.length; ++ci) {
+                for(; gh < ghosts.length; ++gh) {
+                for(; ar < artifacts.length; ++ar) {
+                    var armor = {helmet: helms[h], gauntlets: gaunts[g], chest: chests[c], leg: legs[l], classItem: classItems[ci], ghost: ghosts[gh], artifact: artifacts[ar]};
+                    if(validSet(armor)) {
+                        var set = {armor: armor};
+                        set.int_val = load_stats(set, 144602215, 'int');
+                        set.disc_val = load_stats(set, 1735777505, 'disc');
+                        set.str_val = load_stats(set, 4244567218, 'str');
+                        var int_level = Math.min(Math.floor(set.int_val/60), 5);
+                        var disc_level = Math.min(Math.floor(set.disc_val/60), 5);
+                        var str_level = Math.min(Math.floor(set.str_val/60), 5);
+                        var tiers_string = int_level.toString() + '/' + disc_level.toString() + '/' + str_level.toString();
+                        if(tiers_string in set_map) {
+                            set_map[tiers_string].push(set);
+                        } else {
+                            set_map[tiers_string] = [set];
+                        }
+                        vm.allSetTiers = Object.keys(set_map).sort().reverse();
+                        vm.activesets = vm.allSetTiers[0];
+                    }
+                    
+                    processed_count++;
+                    if((processed_count%200) == 0) {
+                        // If active gaurdian is changed then stop processing combinations
+                        if(vm.active != activeGaurdian) {
+                            return;
+                        }
+                        $scope.$apply(function () {
+                            vm.progress = processed_count/(helms.length * gaunts.length * chests.length * legs.length * classItems.length * ghosts.length * artifacts.length);
+                        });
+                        setTimeout(function() { step(activeGaurdian, h,g,c,l,ci,gh,ar,processed_count); },0);
+                        return;
+                    }
+                } ar = 0; } gh = 0; } ci = 0; } l = 0; } c = 0; } g = 0; }
+                
```

Comment:
moving this out of the loop to here increased the time to calculate all my warlock sets from 24 seconds to 8 seconds!

```
                vm.allSetTiers = Object.keys(set_map).sort().reverse();
                vm.activesets = vm.allSetTiers[0];
```